src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -1028,6 +1029,134 @@ private TwoWindingsTransformer createTwoWindingsTransformer(Network network, Vol
     }
 
+    private List<EquipmenModificationInfos> execCreateVoltageLevel(
+        NetworkStoreListener listener, VoltageLevelCreationInfos voltageLevelCreationInfos,
+        ReporterModel reporter,
+        Reporter subReporter) {
+
+        Network network = listener.getNetwork();
+        UUID networkUuid = listener.getNetworkUuid();
+        Substation substation = network.getSubstation(voltageLevelCreationInfos.getSubstationId());
+
+        return doAction(listener, () -> {
+            VoltageLevel voltageLevel = substation.newVoltageLevel()
+                .setId(voltageLevelCreationInfos.getEquipmentId())
+                .setName(voltageLevelCreationInfos.getEquipmentName())
+                .setTopologyKind(TopologyKind.NODE_BREAKER)
+                .setNominalV(voltageLevelCreationInfos.getNominalVoltage())
+                .add();
+
+            int nodeRank = voltageLevel.getNodeBreakerView().getMaximumNodeIndex() + 1;
+            Map<String, Integer> idToNodeRank = new TreeMap<>();
+            for (BusbarSectionCreationInfos bbs : voltageLevelCreationInfos.getBusbarSections()) {
+                BusbarSection sjb = voltageLevel.getNodeBreakerView().newBusbarSection()
+                    .setId(bbs.getId())
+                    .setName(bbs.getName())
+                    .setNode(nodeRank)
+                    .add();
+                sjb.newExtension(BusbarSectionPositionAdder.class)
+                    .withBusbarIndex(bbs.getVertPos())
+                    .withSectionIndex(bbs.getHorizPos())
+                    .add();
+                idToNodeRank.put(bbs.getId(), nodeRank);
+                nodeRank += 1;
+            }
+
+            int cnxRank = 1;
+            for (BusbarConnectionCreationInfos bbsci : voltageLevelCreationInfos.getBusbarConnections()) {
+                String fromBBSId = bbsci.getFromBBS();
+                Integer rank1 = idToNodeRank.get(fromBBSId);
+                if (rank1 == null) {
+                    throw new NetworkModificationException(CREATE_VOLTAGE_LEVEL_ERROR, "From side '" + fromBBSId + "' unknown");
+                }
+
+                String toBBSId = bbsci.getToBBS();
+                Integer rank2 = idToNodeRank.get(toBBSId);
+                if (rank2 == null) {
+                    throw new NetworkModificationException(CREATE_VOLTAGE_LEVEL_ERROR, "To side '" + toBBSId + "' unknown");
+                }
+
+                SwitchKind switchKind = bbsci.getSwitchKind();
+                if (switchKind == SwitchKind.BREAKER) {
+                    int preBreakerRank = nodeRank++;
+                    int postBreakerRank = nodeRank++;
+                    voltageLevel.getNodeBreakerView().newDisconnector()
+                        .setKind(SwitchKind.DISCONNECTOR)
+                        .setId("switch_" + voltageLevelCreationInfos.getEquipmentId() + "_" + fromBBSId + "_" + toBBSId
```

Comment:
maybe "disconnector_" , like in createNodeBreakerCellSwitches method ?

Reply:
done